dask/dataframe/core.py

```diff
@@ -7060,6 +7043,15 @@ def partitionwise_graph(func, name, *args, **kwargs):
             else:
                 raise ValueError("Can't add multi-dimensional array to dataframes")
             numblocks[arg._name] = arg.numblocks
+        elif isinstance(arg, BlockwiseDep):
+            if len(arg.numblocks) == 1:
+                pairs.extend([arg, "i"])
+            elif len(arg.numblocks) == 2:
+                pairs.extend([arg, "ij"])
+            else:
+                raise ValueError(
+                    f"BlockwiseDep arg {arg!r} has {len(arg.numblocks)} dimensions; only 1 or 2 are supported."
```

Comment:
It seems to me that this section should be using `arg.ndim` rather than `len(arg.numblocks)`. I get that `len(arg.numblocks)` might be faster, but in that case shouldn't `ndim` be rewritten to use that rather than shape?

Reply:
I don't think the `BlockwiseDep` interface has a `ndim` attribute?
https://github.com/dask/dask/blob/5b38dda3ff4c2432baa6af8b52ba1572a539548e/dask/blockwise.py#L35-L98

Reply:
:eyes: I guess I wasn't reading close enough.